app/assets/javascripts/Results/context_menu.js

```diff
@@ -0,0 +1,172 @@
+/*
+  This file contains the Javascript code necessary for the functionality
+  of the annotation context menu. For the annotation context menu to
+  function properly, these dependencies are necessary:
+
+  Context-Menu and library related:
+    * jquery.ui-contextmenu.min.js (..\app\assets\javascripts\)
+      * Depends on jQuery and jQuery-ui
+    * context_menu.css (..\app\assets\stylesheets\)
+
+  Context-Menu button functionality:
+    * _annotations.js.erb (..\app\views\results\common\)
+
+   */
+var annotation_context_menu = {
+  setup: function(annot_path, result_id, assignment_id, file_dl_path) {
+    var menu_items = {
+      new_annotation: {
+        title: 'New Annotation',
+        cmd: 'new_annotation',
+        action: make_new_annotation,
+        disabled: true
+      },
+      common_annotations: {
+        title: 'Common Annotations <kbd>></kbd>',
+        cmd: 'common_annotations',
+        disabled: true
+      },
+      delete_annotation: {
+        title: 'Delete Annotation',
+        cmd: "delete_annotation",
+        action: function(event, ui) {
+          var clicked_element = $(ui.target);
+          var annot_id = '';
+          var sub_file_id = $('#select_file_id').val();
+          if (annotation_type === ANNOTATION_TYPES.CODE) {
+            $.each(clicked_element[0].attributes, function(index, attr) {
+              if (attr.nodeName.toLowerCase()
+                      .indexOf('data-annotationid') != -1) {
+                annot_id = attr.value;
+                // Continue iteration in case of multiple annotations
+              }
+            });
+          } else {
+            annot_id = clicked_element.attr('id')
+                                      .replace('annotation_holder_', '');
+          }
+
+          if (annot_id !== null && annot_id.length !== 0 &&
+              sub_file_id !== null && sub_file_id.length !== 0) {
+            var query_str_params = [
+              ['id=' + annot_id],
+              ['submission_file_id=' + sub_file_id],
+              ['result_id=' + result_id],
+              ['assignment_id=' + assignment_id]
+            ];
+            $.ajax({
+              url: annot_path + '?' + query_str_params.join('&'),
+              method: 'POST',
```

Comment:
Hmm this is strange. When I first implemented this I initially passing the necessary parameters through the post body with the `data` property, but it did not work properly. This was the reason why I was passing them as query string parameters instead (since this was how it was through the delete button). 

However, it seems to be working now when I use method `DELETE`, and passing the arguments through `data`. Thanks for prompting me to check this.